opentelemetry/src/trace/noop.rs

```diff
@@ -38,25 +38,11 @@ pub struct NoopSpan {
     span_context: trace::SpanContext,
 }
 
-impl Default for NoopSpan {
-    fn default() -> Self {
-        NoopSpan::new()
-    }
-}
-
 impl NoopSpan {
-    /// Creates a new `NoopSpan` instance.
-    pub fn new() -> Self {
-        NoopSpan {
-            span_context: trace::SpanContext::new(
-                trace::TraceId::INVALID,
-                trace::SpanId::INVALID,
-                TraceFlags::default(),
-                false,
-                TraceState::default(),
-            ),
-        }
-    }
+    /// The default `NoopSpan`, as a constant
+    pub const DEFAULT: NoopSpan = NoopSpan {
```

Comment:
I suppose we could use an inherent `const fn default()` here instead of the associated const. On the one hand, that would look more familiar, on the other hand maybe it's an uncanney valley trap?